tests/python_package_test/test_engine.py

```diff
@@ -48,2591 +49,2655 @@ def categorize(continuous_x):
+    X_train, y_train = load_svmlight_file(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                       '../../examples/lambdarank/rank.train'))
+    q_train = np.loadtxt(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                      '../../examples/lambdarank/rank.train.query'))
+    params_lambdarank = {'objective': 'lambdarank', 'verbose': -1, 'eval_at': 3}
+    lgb_train = lgb.Dataset(X_train, y_train, group=q_train)
+    # ... with l2 metric
+    cv_res_lambda = lgb.cv(params_lambdarank, lgb_train, num_boost_round=10, nfold=3,
+                           metrics='l2', verbose_eval=False)
+    assert len(cv_res_lambda) == 2
+    assert not np.isnan(cv_res_lambda['l2-mean']).any()
+    # ... with NDCG (default) metric
+    cv_res_lambda = lgb.cv(params_lambdarank, lgb_train, num_boost_round=10, nfold=3,
+                           verbose_eval=False)
+    assert len(cv_res_lambda) == 2
+    assert not np.isnan(cv_res_lambda['ndcg@3-mean']).any()
+    # self defined folds with lambdarank
+    cv_res_lambda_obj = lgb.cv(params_lambdarank, lgb_train, num_boost_round=10,
+                               folds=GroupKFold(n_splits=3),
+                               verbose_eval=False)
+    np.testing.assert_allclose(cv_res_lambda['ndcg@3-mean'], cv_res_lambda_obj['ndcg@3-mean'])
+
+
+def test_cvbooster():
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+    }
+    lgb_train = lgb.Dataset(X_train, y_train)
+    # with early stopping
+    cv_res = lgb.cv(params, lgb_train,
+                    num_boost_round=25,
+                    early_stopping_rounds=5,
+                    verbose_eval=False,
+                    nfold=3,
+                    return_cvbooster=True)
+    assert 'cvbooster' in cv_res
+    cvb = cv_res['cvbooster']
+    assert isinstance(cvb, lgb.CVBooster)
+    assert isinstance(cvb.boosters, list)
+    assert len(cvb.boosters) == 3
+    assert all(isinstance(bst, lgb.Booster) for bst in cvb.boosters)
+    assert cvb.best_iteration > 0
+    # predict by each fold booster
+    preds = cvb.predict(X_test, num_iteration=cvb.best_iteration)
+    assert isinstance(preds, list)
+    assert len(preds) == 3
+    # fold averaging
+    avg_pred = np.mean(preds, axis=0)
+    ret = log_loss(y_test, avg_pred)
+    assert ret < 3
+    # without early stopping
+    cv_res = lgb.cv(params, lgb_train,
+                    num_boost_round=20,
+                    verbose_eval=False,
+                    nfold=3,
+                    return_cvbooster=True)
+    cvb = cv_res['cvbooster']
+    assert cvb.best_iteration == -1
+    preds = cvb.predict(X_test)
+    avg_pred = np.mean(preds, axis=0)
+    ret = log_loss(y_test, avg_pred)
+    assert ret < 5
```

Comment:
```suggestion
    assert ret < 0.15
```

https://www.textcompare.org/index.html?id=6008e19c6d59840017ac7a56

![image](https://user-images.githubusercontent.com/7608904/105271147-9df30380-5b5c-11eb-9038-e44c14083cc8.png)